### PR TITLE
Convert crash on binop to controlled error

### PR DESCRIPTION
**Pull request by septract**
Partially addresses #233

### CODE REVIEW

**Comment by septract** [comment]
@cp526 I guessed at an error message, but perhaps you can propose something better?

File: backend/cn/lib/check.ml [hunk 1]
```diff
@@ -693,10 +693,15 @@ let rec check_pexpr (pe : BT.t mu_pexpr) (k : IT.t -> unit m) : unit m =
...
-       assert (
+       let@ () =
          match act.ct with
-         | Integer ity when Sctypes.is_unsigned_integer_type ity -> true
-         | _ -> false);
+         | Integer ity when Sctypes.is_unsigned_integer_type ity -> return ()
+         | _ ->
+           fail (fun _ ->
+             { loc;
+               msg = Generic !^"M_PEbounded_binop: unsupported on non-integer types"
```
Comment (cp526) [inline backend/cn/lib/check.ml:702]: This should probably mention that this is unsupported on anything other than *unsigned* integers.

**Comment by cp526** [comment]
... Although I think these might really be proper assertion failures.

**Comment by septract** [comment]
What does it mean for this to be a proper assertion failure? I would assume these should only occur in anomalous states, ie states that should be prevented by some other CN code

**Comment by cp526** [comment]
I suspect we're in that kind of situation here: something in CN+Cerberus's earlier logic should prevent an application of `M_PEbounded_binop (M_Bound_Wrap, ...)` when the C-type is a signed integer (wrapping behaviour typically only happens for unsigned types).

Similarly, the M_PEbounded_binop below should only happen on integer types.

**Comment by septract** [comment]
Huh, okay - so is the conclusion that the crash in #233 is actually a bug in CN, rather than an unsupported feature? That seems more difficult to resolve, & probably needs you to take a deeper look @cp526

**Comment by cp526** [comment]
Yes, I think so; I see no reason why CN shouldn't handle the example from #233 .

**Comment by septract** [comment]
Okay, I'm closing this PR - looks like someone should look at the root cause of #233 (I'll take a look but I suspect it's too deep in the stack for my level of understanding)

**Pull request closed without merging**